src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,120 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Node<K,V>[] nodeArray;
+    private int size;
+    private int capacity;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        nodeArray = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        capacity = INITIAL_CAPACITY;
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        size += putNode(nodeArray, new Node<>(key, value), false);
+        if (size >= nodeArray.length * LOAD_FACTOR) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        for (int i = 0; i < nodeArray.length; i++) {
+            Node<K,V> node = nodeArray[i];
+            if (node == null) {
+                continue;
+            }
+            do {
+                if (keysEqual(node.key, key)) {
+                    return node.value;
+                }
+                node = node.next;
+            } while (node != null);
+        }
         return null;
```

Comment:
can be simplified. Use your gethash method instead of for loop, while instead of do while, try to do smth like this
![image](https://github.com/mate-academy/jv-hash-map/assets/49767049/aa9c8972-94f3-4687-a3e5-f6f49f10b15a)

Reply:
right. thanks!